pkg/sysregistries/system_registries.go

```diff
@@ -29,8 +30,12 @@ type tomlConfig struct {
 
 // Reads the global registry file from the filesystem. Returns
 // a byte array
-func readRegistryConf() ([]byte, error) {
-	configBytes, err := ioutil.ReadFile(systemRegistriesConfPath)
+func readRegistryConf(ctx *types.SystemContext) ([]byte, error) {
+	dirPath := systemRegistriesConfPath
+	if ctx != nil {
+		dirPath = ctx.SystemRegistriesConfPath
+	}
```

Comment:
This must also check for `ctx.SystemRegistriesConfPath != ""`; a `SystemContext` may be provided by the caller to override something else in a deep call stack.

Please include support for `RootForImplicitAbsolutePaths` as well.

Reply:
I think if caller sets to "" it's his fault, maybe this should be a `*string` instead so we can check if caller sets it?

Reply:
Golang does not have optionals, a `string` is `""` by default, it’s not something the caller does intentionally.

A `*string` would work as well here, but `""` is a clearly invalid value for a path, and all the other users treat `""` as unspecified as well.

(The pedantic me would _love_ to have true optionals in the language, and to treat `""` as any other string value WRT code paths. But, we have Golang to work with, and all of the other users doing it strictly-speaking-wrong, so let’s at least be consistently wrong/ugly :-| )

Reply:
> Golang does not have optionals, a string is "" by default, it’s not something the caller does intentionally.

I know, that's why (following opencontainers specs) I suggested `*string`. If it's `nil`,caller didn't set it, otherwise, it's her fault if she passed an empty string.

Reply:
Added

Reply:
`*string` would be fine with me, sure. It _is_ strictly speaking better, and we might just as well start somewhere.

(I can live with `""` as well.)